Update typography guidance and restructure styles section (#3343)
alphagov/govuk-design-system: src/styles/type-scale/index.md

---
title: Type scale
description: The type scale underpins all of the text styles on GOV.UK

> section: Styles

Author: ```suggestion
section: Styles
backlogIssueId: 64
```

theme: Typography
layout: layout-pane.njk
order: 6
---

{% from "_example.njk" import example %}

{% from "govuk/components/details/macro.njk" import govukDetails %}
{% from "govuk/components/table/macro.njk" import govukTable %}

> The type scale is a collection of font sizes and line heights that underpin all of the typography styles on GOV.UK. It has been tested and [iterated](https://designnotes.blog.gov.uk/2022/12/12/making-the-gov-uk-frontend-typography-scale-more-accessible/) for readability on different devices.

Author: ```suggestion
The type scale is a collection of font sizes and line heights that underpin all of the typographic styles on GOV.UK. It has been tested and [iterated](https://designnotes.blog.gov.uk/2022/12/12/making-the-gov-uk-frontend-typography-scale-more-accessible/) for readability on different devices.
```

      {
        text: "27px",
        format: "numeric"
      },
      {
        text: "30px",
        format: "numeric"
      }
    ],
    [
      {
        text: "27"
      },
      {
        text: "Only used in exceptional circumstances"
      },
      {
        text: "21px",
        format: "numeric"
      },
      {
        text: "25px",
        format: "numeric"
      }
    ],
    [
      {
        text: "24"
      },
      {
        html: "<code>govuk-heading-m</code>, <code>govuk-body-l</code>"
      },
      {
        text: "21px",
        format: "numeric"
      },
      {
        text: "25px",
        format: "numeric"
      }
    ],
    [
      {
        text: "19"
      },
      {
        html: "<code>govuk-heading-s</code>, <code>govuk-body</code>"
      },
      {
        text: "19px",
        format: "numeric"
      },
      {
        text: "25px",
        format: "numeric"
      }
    ],
    [
      {
        text: "16"
      },
      {
        html: "<code>govuk-body-s<code>"
      },
      {
        text: "16px",
        format: "numeric"
      },
      {
        text: "20px",
        format: "numeric"
      }
    ]
  ]
}) }}

### Updates to the type scale

> In 2024, we updated the type scale to make text easier to read on smaller screens. We published this change in v5.x of GOV.UK Frontend.

Author: ```suggestion
In 2024, we updated the type scale to make text easier to read on smaller screens. We published this change in v5.2 of GOV.UK Frontend. Read more about [updating your service to use the new type scale](/get-started/new-typography-scale/)
```

> <details class="govuk-details">

Reviewer: I think we should use a details macro here if we can. We can `set` the contents of the detail.

  <summary class="govuk-details__summary">
    <span class="govuk-details__summary-text">

>       View the old typography scale

Author: ```suggestion
      View the old type scale
```

    </span>
  </summary>
  <div class="govuk-details__text">

>     {{ govukTable({

Reviewer: 2 things:
- We need another table to communicate responsive behaviour like we have above
- We need 14 on here to show that it existed on the old type scale (should we show this?)

          [
            {
              text: "24"
            },
            {
              html: "<code>govuk-heading-m</code>, <code>govuk-body-l</code>"
            },
            {
              text: "21px",
              format: "numeric"
            },
            {
              text: "25px",
              format: "numeric"
            }
          ],
          [
            {
              text: "19"
            },
            {
              html: "<code>govuk-heading-s</code>, <code>govuk-body</code>"
            },
            {
              text: "19px",
              format: "numeric"
            },
            {
              text: "25px",
              format: "numeric"
            }
          ],
          [
            {
              text: "16"
            },
            {
              html: "<code>govuk-body-s<code>"
            },
            {
              text: "16px",
              format: "numeric"
            },
            {
              text: "20px",
              format: "numeric"
            }
          ]
        ]
      }) }}

      The scale for large screens has not changed.
  </div>
</details>

## Using the type scale in your own CSS

> Include the `govuk-font` mixin to use GOV.UK typography in your CSS. 

Reviewer: Could we link to the [API docs](https://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-font) here?

Reviewer: Decision: do it!

For example, if you want a custom element to use type scale unit 19, use:

```scss
@include govuk-font($size: 19);
``````

You can add additional arguments to control font weight, use tabular font spacing, or to override the line height:

```scss
@include govuk-font($size 19, $weight: bold, $tabular: true);
```

### If you only want to set the font size

> Do not use `govuk-font` if you only want to change the font size and line height – it includes additional typography-related CSS like the New Transport font family. Instead, you should use `govuk-font-size`:

Reviewer: Same thing for govuk-font-size (we changed the name so the link to the API docs doesn't exist currently but it will after 5.1 goes out).
